### DISCUSSION TITLE
[Urgent] Unable to launch the iced application on Window Server 2012 Error: GraphicsAdapterNotFound

### DISCUSSION THREAD

**Discussion by mayankjoshii** [hecrj/iced#1959] [Q&A]
I have an application consuming the iced library for GUI. The GUI is launching perfectly in all the Operation Systems except for the **Window Sever 2012**.  There it is exiting with the error :  `GraphicsCreationFailed<GraphicsAdapterNotFound>`

I also tried the workaround provided by iced library in official Documentation to try building the with the `iced_glow` render instead of `iced_wgpu` renderer, but still no help. 

If I am building my application with iced/glow feature then I'm getting separate error `GraphicsCreationFailed(NoAvailablePixelFormat)`, and UI screen is launching twice and freezing. Also, with using glow as a renderer the GUI is not launching in any OS. 

What are the other option to fix this? If it's some renderer issues, can we carry some dependent .dll to fix this issue, or anything else that can be done to fix this problem?

Please help as this is urgent and blocking the release.

**Comment by genusistimelord** [comment]
this happens due to servers generally not having graphics cards to begin with. You can wait for iced to get the pixelbuffer stuff implemented.. That should fix your issue but I'm not sure if its in the main branch or not yet. If you are running it in a VM then you need to make sure your allowing graphics to pass thru to it and install the needed drivers. If you do have a GPU installed on the machine you will need to install the drivers. A lot of servers really don't support much in graphics.

**Reply by mayankjoshii** [reply]
I was reading the iced docs, and found out that iced has rem**oved the support for DirectX 11 (DX11)**. 

In all the system where the APP is launching successfully have the **DirectX 12 (DX12)** but **windows 2012** server has the **DirectX 11** (DX11). Do you believer this is the issue?

 

> You can wait for iced to get the pixelbuffer stuff implemented.. That should fix your issue but I'm not sure if its in the main branch or not yet. 

Any ETA on when this will be done? Also, is this implementation will be done for the `iced_glow` launch failure.

**Comment by AustinMReppert** [comment]
I thought the tiny skia renderer was cpu only. Might want to give it a try.

**Comment by genusistimelord** [comment]
yeah I'm guessing Tiny Skia is the CPU renderer.
